diff -up db-18.1.25/src/repmgr/repmgr_net.c.omv~ db-18.1.25/src/repmgr/repmgr_net.c
--- db-18.1.25/src/repmgr/repmgr_net.c.omv~	2019-02-15 05:41:43.470940278 +0100
+++ db-18.1.25/src/repmgr/repmgr_net.c	2019-02-15 05:42:40.260777823 +0100
@@ -2894,6 +2894,7 @@ err_cleanup:
 	if (ssl_ctx)
 		SSL_CTX_free(ssl_ctx);
 
+#if defined(OPENSSL_THREADS) && OPENSSL_VERSION_NUMBER < 0x10100000L
 	if (repmgr_ssl_mutex_arr != NULL) {
 		for (i = 0; i < CRYPTO_num_locks(); i++) {
 			if (repmgr_ssl_mutex_arr[i] != NULL)
@@ -2902,6 +2903,7 @@ err_cleanup:
 		
 		repmgr_ssl_mutex_arr = NULL;
 	}
+#endif
 
 	return (1);
 }
